src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -226,14 +227,36 @@ private void addOptions(Template template, Options options, String infrastructur
         if (securityGroupNames != null && !securityGroupNames.isEmpty()) {
             template.getOptions().as(AWSEC2TemplateOptions.class).securityGroupIds(securityGroupNames);
         } else {
-            usingAutoGeneratedSecurityGroupInfrastructures.add(infrastructureId);
+
             log.info(String.format("The infrastructure [%s] is using the auto-generated security group.",
-                                   infrastructureId));
+                                   infrastructure.getId()));
             // Have we defined an explicit list of ports to be be opened ?
-            Optional<int[]> portsToBeExplicitlyOpened = Optional.ofNullable(options.getPortsToOpen());
-            portsToBeExplicitlyOpened.ifPresent(integers -> template.getOptions()
-                                                                    .as(AWSEC2TemplateOptions.class)
-                                                                    .inboundPorts(integers));
+            int[] ports = options.getPortsToOpen();
+            String sgName = getAutoGeneratedSecurityGroupName(infrastructure.getId());
+            if (ports != null) {
+                sgName += "_" + UUID.randomUUID();
+                String sgDescription = "Auto generated security group to authorize the ports " + Arrays.toString(ports);
+                SecurityGroupApi sgApi = getSecurityGroupApi(infrastructure);
+                sgApi.createSecurityGroupInRegion(region, sgName, sgDescription);
+                for (int port : ports) {
+                    sgApi.authorizeSecurityGroupIngressInRegion(region,
+                                                                sgName,
+                                                                IpProtocol.TCP,
```

Comment:
- I guess IpProtocol.ALL could prevent specifying twice SG rules to address both UDP & TCP
- If port == -1, then you need to go to IpProtocol.ICMP

Reply:
After a check, actuallly IpProtocol.ALL means opening all ports.
And I'll add ICMP for -1.

ref:
- https://github.com/hashicorp/terraform/issues/1177#issuecomment-98798247
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-security-group-ingress.html